stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/OTPElementUI.kt

```diff
@@ -120,77 +124,94 @@ fun OTPElementUI(
                     textFieldModifier = textFieldModifier.focusRequester(focusRequester)
                 }
 
-                // Need to use BasicTextField instead of TextField to be able to customize the
-                // internal contentPadding
-                BasicTextField(
-                    value = TextFieldValue(
-                        text = value,
-                        selection = if (isSelected) {
-                            TextRange(value.length)
-                        } else {
-                            TextRange.Zero
-                        }
-                    ),
-                    onValueChange = {
-                        val inputLength = element.controller.onValueChanged(index, it.text)
-                        (0 until inputLength).forEach { _ ->
-                            focusManager.moveFocus(FocusDirection.Next)
-                        }
-                    },
-                    modifier = textFieldModifier,
+                OTPInputBox(
+                    value = value,
+                    isSelected = isSelected,
+                    element = element,
+                    index = index,
+                    focusManager = focusManager,
+                    textFieldModifier = textFieldModifier,
```

Comment:
Boy Scout rule here - fixed the `LongMethod` report extracting the `OTPInputBox` to its own Composable function.